resources/todomvc/architecture-examples/preact/src/app/item.js

```diff
@@ -1,85 +1,49 @@
+        const val = e.target.value.trim();
         if (val) {
             onSave(todo, val);
-            this.setState({ editText: val });
+            setEditing(false);
+        } else {
+            onRemove(todo);
         }
-        else {
-            onDestroy(todo);
-        }
-    };
+    }
 
-    handleEdit = () => {
-        let { onEdit, todo } = this.props;
-        onEdit(todo);
-        this.setState({ editText: todo.title });
-    };
+    function handleKeyDown(e) {
+        // prettier-ignore
+        if (e.key === "Escape" || e.key === "ESCAPE")
+            setEditing(false);
+        else if (e.key === "Enter" || e.key === "ENTER")
+            handleSubmit(e);
+    }
+
+    function handleDoubleClick() {
+        setEditing(true);
+    }
 
-    toggle = e => {
-        let { onToggle, todo } = this.props;
+    function handleToggle(e) {
         onToggle(todo);
         e.preventDefault();
-    };
-
-    handleKeyDown = e => {
-        if (e.which===ESCAPE_KEY) {
-            let { todo } = this.props;
-            this.setState({ editText: todo.title });
-            this.props.onCancel(todo);
-        }
-        else if (e.which===ENTER_KEY) {
-            this.handleSubmit();
-        }
-    };
-
-    handleDestroy = () => {
-        this.props.onDestroy(this.props.todo);
-    };
-
-    // shouldComponentUpdate({ todo, editing, editText }) {
-    //  return (
-    //      todo !== this.props.todo ||
-    //      editing !== this.props.editing ||
-    //      editText !== this.state.editText
-    //  );
-    // }
-
-    componentDidUpdate() {
-        let node = this.base && this.base.querySelector('.edit');
-        if (node) node.focus();
     }
 
-    render({ todo:{ title, completed }, onToggle, onDestroy, editing }, { editText }) {
-        return (
-            <li class={cx({ completed, editing })}>
-                <div class="view">
-                    <input
-                        class="toggle"
-                        type="checkbox"
-                        checked={completed}
-                        onChange={this.toggle}
-                    />
-                    <label onDblClick={this.handleEdit}>{title}</label>
-                    <button class="destroy" onClick={this.handleDestroy} />
-                </div>
-                { editing && (
-                    <input
-                        class="edit"
-                        value={editText}
-                        onBlur={this.handleSubmit}
-                        onInput={this.linkState('editText')}
-                        onKeyDown={this.handleKeyDown}
-                    />
-                ) }
-            </li>
-        );
+    function handleRemove() {
+        onRemove(todo);
     }
+
+    return (
+        <li class={cx({ completed: todo.completed, editing })}>
+            <div class="view">
+                <input class="toggle" type="checkbox" checked={todo.completed} onChange={handleToggle} />
```

Comment:
yes - uncontrolled on purpose. 
There was a slight increase in performance when I removed local state.